registrations/tasks.py

```diff
@@ -375,6 +382,118 @@ def run(self, registration_id, **kwargs):
+                back to ndoh-control's "authority" fields to maintain
+                backwards compatibility with existing APIs
+        """
+        return {
+            'CLINIC USSD app': 'clinic',
+            'PUBLIC USSD app': 'personal',
+            'CHW USSD app': 'chw',
+            'NURSE USSD App': 'nurseconnect',
+        }.get(source.name, 'unknown')  # See note at get_subscription_type
+
+    def build_jembi_json(self, registration):
+        """ Compile json to be sent to Jembi. """
+        authority = self.get_authority_from_source(registration.source)
+        json_template = {
+            "mha": 1,
+            "swt": 1,
+            "dmsisdn": registration.data['msisdn_device'],
+            "cmsisdn": registration.data['msisdn_registrant'],
+            "id": self.get_patient_id(
+                registration.data['id_type'],
+                (registration.data['sa_id_no']
+                 if registration.data['id_type'] == 'sa_id'
+                 else registration.data['passport_no']),
+                # passport_origin may be None if sa_id is used
+                registration.data.get('passport_origin'),
+                registration.data['msisdn_registrant']),
+            "type": self.get_subscription_type(authority),
+            "lang": registration.data['language'],
+            "encdate": self.get_timestamp(),
+            "faccode": registration.data['faccode'],
+            "dob": self.get_dob(
+                datetime.strptime(registration.data['mom_dob'], '%Y-%m-%d'))
+        }
+
+        # Self registrations on all lines should use cmsisdn as dmsisdn too
+        if registration.data['msisdn_device'] is None:
+            json_template["dmsisdn"] = registration.data['msisdn_registrant']
+
+        if authority == 'clinic':
+            json_template["edd"] = registration.data["edd"].strftime("%Y%m%d")
+
+        return json_template
+
+    def run(self, registration_id, **kwargs):
+        self.l.info("Compiling Jembi Json data")
+        from .models import Registration
+        registration = Registration.objects.get(pk=registration_id)
+        json_doc = self.build_jembi_json(registration)
+        try:
+            result = requests.post(
+                "%s/subscription" % settings.JEMBI_BASE_URL,  # url
+                headers={'Content-Type': 'application/json'},
+                data=json.dumps(json_doc),
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                verify=False
+            )
+            result.raise_for_status()
+            return result.text
+        except (HTTPError,) as e:
+            # retry message sending if in 500 range (3 default retries)
+            if 500 < e.response.status_code < 599:
+                raise self.retry(exc=e)
+            else:
+                raise e
+        except (Exception,) as e:
+            self.l.error('Problem posting JSON to Jembi', exc_info=True)
```

Comment:
Perhaps we can include the registration uuid at this point in case we need to rerun this manually.